File: platforms/hacker-rank/E--15--day-of-the-programmer/solutions/solution.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'dayOfProgrammer' function below.
#
# The function is expected to return a STRING.
# The function accepts INTEGER year as parameter.
#


def getDays(year):
    days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    if year == 1918:
        days[1] -= 13
    elif year < 1918:
        if year % 4 == 0:
            days[1] = 29
    else:
        if (year % 400 == 0) or ((year % 4 == 0) and (year % 100 != 0)):
            days[1] = 29
    return days


def dayOfProgrammer(year):
    days = getDays(year)
    if sum(days) == 352:
        return "26.09.1918"
    elif sum(days) == 366:
        return f"12.09.{year}"
    else:
        return f"13.09.{year}"


if __name__ == "__main__":
    fptr = open(os.environ["OUTPUT_PATH"], "w")
    year = int(input().strip())
    result = dayOfProgrammer(year)
    fptr.write(result + "\n")
    fptr.close()

File: platforms/hacker-rank/E--15--day-of-the-programmer/solutions/solution.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"strconv"
	"strings"
)

/*
 * Complete the 'dayOfProgrammer' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts INTEGER year as parameter.
 */

func totDays(year int32) int32 {
	var days int32 = 365

	if year == 1918 {
		days -= 13
	} else if year < 1918 {
		if year%4 == 0 {
			days++
		}
	} else {
		if year%400 == 0 {
			days++
		} else if year%4 == 0 && year%100 != 0 {
			days++
		}
	}

	return days
}

func dayOfProgrammer(year int32) string {
	days := totDays(year)
	if days == 352 {
		return "26.09.1918"
	} else if days == 366 {
		return fmt.Sprintf("12.09.%v", year)
	} else {
		return fmt.Sprintf("13.09.%v", year)
	}
}

func main() {
	reader := bufio.NewReaderSize(os.Stdin, 16*1024*1024)

	stdout, err := os.Create(os.Getenv("OUTPUT_PATH"))
	checkError(err)

	defer stdout.Close()

	writer := bufio.NewWriterSize(stdout, 16*1024*1024)

	yearTemp, err := strconv.ParseInt(strings.TrimSpace(readLine(reader)), 10, 64)
	checkError(err)
	year := int32(yearTemp)

	result := dayOfProgrammer(year)

	fmt.Fprintf(writer, "%s\n", result)

	writer.Flush()
}

func readLine(reader *bufio.Reader) string {
	str, _, err := reader.ReadLine()
	if err == io.EOF {
		return ""
	}

	return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
	if err != nil {
		panic(err)
	}
}
